utils/gitlab_api.py

```diff
@@ -0,0 +1,97 @@
+import gitlab
+import urllib3
+import uuid
+from reconcile.config import get_config
+
+
+class GitLabApi(object):
+    def __init__(self, project_name, ssl_verify=False):
+        # The following line will supress
+        # `InsecureRequestWarning: Unverified HTTPS request is being made`
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+        self.project = init_project_from_config(project_name, ssl_verify)
+
+    def create_branch(self, new_branch, source_branch):
+        data = {
+            'branch': new_branch,
+            'ref': source_branch
+        }
+        self.project.branches.create(data)
+
+    def delete_file(self, branch_name, file_path, commit_message):
+        data = {
+            'branch': branch_name,
+            'commit_message': commit_message,
+            'actions': [
+                {
+                    'action': 'delete',
+                    'file_path': file_path
+                    }
+            ]
+        }
+        self.project.commits.create(data)
+
+    def create_mr(self, source_branch, target_branch, title,
+                  remove_source_branch=True):
+        data = {
+            'source_branch': source_branch,
+            'target_branch': target_branch,
+            'title': title,
+            'remove_source_branch': str(remove_source_branch)
+        }
+        self.project.mergerequests.create(data)
+
+    def delete_user_mr_exists(self, title):
+        mrs = self.project.mergerequests.list(state='opened')
+        for mr in mrs:
+            # since we are using a naming convention for these MRs
+            # we can determine if a pending MR exists based on the title
+            if (mr.attributes['title'] != title):
+                continue
+
+            return True
+
+        return False
+
+    def create_delete_user_mr(self, username, path):
+        prefix = 'qcontract-reconcile'
+        target_branch = 'master'
+        branch_name = '{}-delete-{}-{}'.format(
+            prefix,
+            username,
+            str(uuid.uuid4())[0:6]
+        )
+        title = '[{}] delete user {} ({})'.format(prefix, username, path)
+
+        if (self.delete_user_mr_exists(title)):
+            return
+
+        self.create_branch(branch_name, target_branch)
+        self.delete_file(branch_name, path, title)
+        self.create_mr(branch_name, target_branch, title)
+
+
+def init_project_from_config(project_name, ssl_verify):
```

Comment:
instead of this, I think I would use a method like:

`def get_app_interface_gitlab_api`

which returns a new instance of the `GitLabApi`

using the config file